src/components/TableDialog.vue

```diff
@@ -157,6 +187,88 @@ export default Vue.extend({
         this.fileName = this.fileName || getFileName(file);
         this.fileUploadError = null;
       }
+
+      interface TypeScore {
+        strings: Set<string>;
+        number: number;
+        date: number;
+        total: number;
+      }
+      const columnTypes = new Map<string, TypeScore>();
+
+      Papa.parse(file, {
+        header: true,
+        skipEmptyLines: true,
+        step(row: ParseResult<{}>) {
+          // eslint-disable-next-line @typescript-eslint/no-explicit-any
+          const data = row.data as { [key: string]: any };
+
+          Object.keys(data).forEach((key: string) => {
+            if (!columnTypes.has(key)) {
+              columnTypes.set(key, {
+                strings: new Set<string>(),
+                number: 0,
+                date: 0,
+                total: 0,
+              });
+            }
+
+            const entry = columnTypes.get(key);
+            if (entry === undefined) {
+              throw new Error('impossible');
+            }
```

Comment:
If you're only throwing an error here to satisfy TypeScript, and you're okay with betting that this is actually impossible, you can do the following:

```suggestion
            // eslint-disable-next-line @typescript-eslint/no-non-null-assertion
            const entry = columnTypes.get(key)!;
```

Reply:
Even though mine is more verbose, I think I'll keep it. In this case, using the non-null assertion operator makes sense, but I do think it's somewhat harder to read, and the noise caused by having to suppress a rule seems to also go against legibility.